crates/matrix-sdk-base/src/rooms/mod.rs

```diff
@@ -195,9 +195,36 @@ impl BaseRoomInfo {
                 let mut o_ev = o_ev.clone();
                 o_ev.content.set_created_ts_if_none(o_ev.origin_server_ts);
 
+                let state_key = m.state_key();
+                let owned_user_id = match UserId::parse(state_key) {
```

Comment:
1. Can you extract this into a helper function please?
2. Can you add unit tests for this function please?

Reply:
[`20b5dba`](https://github.com/matrix-org/matrix-rust-sdk/pull/3836/commits/20b5dba3e0b19d312c8df4e6987c46cdf742be5a) & [`cfbe204`](https://github.com/matrix-org/matrix-rust-sdk/pull/3836/commits/cfbe204313f4760f8375efaf8bdb1bb2b3cda4d5)